utils/analysis/analysis.go

```diff
@@ -78,17 +81,16 @@ func IsWorse(current, new v1alpha1.AnalysisStatus) bool {
 	return newIndex > currentIndex
 }
 
-// IsTerminating returns whether or not the analysis run is terminating
+// IsTerminating returns whether or not the analysis run is terminating, either because a terminate
+// was requested explicitly, or because a metric has already measured Failed, Error, or Inconclusive
+// which causes the run to end prematurely.
 func IsTerminating(run *v1alpha1.AnalysisRun) bool {
-	return run.Spec.Terminate || IsFailing(run)
-}
-
-// IsFailing returns whether or not any metric has measured Failed or Error, which will eventually
-// cause the entire run to fail.
-func IsFailing(run *v1alpha1.AnalysisRun) bool {
+	if run.Spec.Terminate {
+		return true
+	}
 	for _, res := range run.Status.MetricResults {
```

Comment:
Can we add a nil check for run.status? This will be mainly helpful for the Rollout and Experiment implementation if the analysis hasn't started yet